test/mock-agent.js

```diff
@@ -772,6 +773,334 @@ test('MockAgent - should persist requests', async (t) => {
   }
 })
 
+test('MockAgent - getCallHistory with no name parameter should return the global call history', async (t) => {
+  t = tspl(t, { plan: 2 })
+
+  const mockAgent = new MockAgent()
+  setGlobalDispatcher(mockAgent)
+  after(() => mockAgent.close())
+
+  const mockClient = mockAgent.get('http://localhost:9999')
+  mockClient.intercept({
+    path: '/foo',
+    method: 'GET'
+  }).reply(200, 'foo')
+
+  t.strictEqual(MockCallHistory.AllMockCallHistory.size, 1)
+  t.ok(mockAgent.getCallHistory() instanceof MockCallHistory)
+})
+
+test('MockAgent - getCallHistory with name parameter should return the named call history', async (t) => {
+  t = tspl(t, { plan: 2 })
+
+  const mockAgent = new MockAgent()
+  setGlobalDispatcher(mockAgent)
+  after(() => mockAgent.close())
+
+  const mockClient = mockAgent.get('http://localhost:9999')
+  mockClient.intercept({
+    path: '/foo',
+    method: 'GET'
+  }).reply(200, 'foo').registerCallHistory('my-history')
+
+  t.strictEqual(MockCallHistory.AllMockCallHistory.size, 2)
+  t.ok(mockAgent.getCallHistory('my-history') instanceof MockCallHistory)
+})
+
+test('MockAgent - getCallHistory with name parameter should return undefined when unknown name history', async (t) => {
+  t = tspl(t, { plan: 2 })
+
+  const mockAgent = new MockAgent()
+  setGlobalDispatcher(mockAgent)
+  after(() => mockAgent.close())
+
+  const mockClient = mockAgent.get('http://localhost:9999')
+  mockClient.intercept({
+    path: '/foo',
+    method: 'GET'
+  }).reply(200, 'foo').registerCallHistory('my-history')
+
+  t.strictEqual(MockCallHistory.AllMockCallHistory.size, 2)
+  t.strictEqual(mockAgent.getCallHistory('no-exist'), undefined)
+})
+
+test('MockAgent - getCallHistory with name parameter should throw when a named history with the name parameter already exist', async (t) => {
+  t = tspl(t, { plan: 1 })
+
+  const mockAgent = new MockAgent()
+  setGlobalDispatcher(mockAgent)
+  after(() => mockAgent.close())
+
+  const mockClient = mockAgent.get('http://localhost:9999')
+  mockClient.intercept({
+    path: '/foo',
+    method: 'GET'
+  }).reply(200, 'foo').registerCallHistory('my-history')
+
+  t.throws(() => mockClient.intercept({
+    path: '/bar',
+    method: 'POST'
+  }).reply(200, 'bar').registerCallHistory('my-history'), new InvalidArgumentError('a CallHistory with name my-history already exist'))
```

Comment:
Shall we throw? Wouldn't be better to overwrite the current story?

Reply:
I think it would be better to use the same history. This will let us using a same history on different interceptor

Reply:
done